sui/token_bridge/sources/bridge_state.move

```diff
@@ -0,0 +1,113 @@
+module token_bridge::bridge_state {
+   //use std::vector::{Self};
+   use std::option::{Self, Option};
+
+   //use sui::object::{UID};
+   //use sui::coin::TreasuryCap;
+   use sui::vec_map::{Self, VecMap};
+   use sui::vec_set::{Self, VecSet};
+
+   use wormhole::external_address::ExternalAddress;
+   use wormhole::myu16::{U16};
+
+   const E_ORIGIN_CHAIN_MISMATCH: u64 = 0;
+   const E_ORIGIN_ADDRESS_MISMATCH: u64 = 1;
+   const E_WRAPPING_NATIVE_COIN: u64 = 2;
+   const E_WRAPPED_ASSET_NOT_INITIALIZED: u64 = 3;
+
+   friend token_bridge::vaa;
+
+   /// The origin chain and address of a token.  In case of native tokens
+   /// (where the chain is aptos), the token_address is the hash of the token
+   /// info (see token_hash.move for more details)
+   struct OriginInfo has store, copy, drop {
+      token_chain: U16,
+      token_address: ExternalAddress,
+   }
+
+   struct BridgeState {
+      governance_chain_id: U16,
+      governance_contract: ExternalAddress,
+
+      /// Set of consumed VAA hashes
+      consumed_vaas: VecSet<vector<u8>>,
```

Comment:
for now this is fine, but we will have to consider a more efficient data structure here before the contracts go live. O(n) operations will become prohibitively expensive quickly

Reply:
agree